pkg/extension/sumologicextension/extension.go

```diff
@@ -200,104 +169,166 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
+	httpClientSettings confighttp.HTTPClientSettings,
+	host component.Host,
+	regInfo api.OpenRegisterResponsePayload,
+) (*http.Client, error) {
+
+	httpClient, err := httpClientSettings.ToClient(
 		host.GetExtensions(),
 		component.TelemetrySettings{},
 	)
 	if err != nil {
-		return credentials.CollectorCredentials{},
-			registrationDone,
-			fmt.Errorf("couldn't create HTTP client: %w", err)
+		return nil, fmt.Errorf("couldn't create HTTP client: %w", err)
 	}
 
-	// Set the transport so that all requests from se.httpClient will contain
+	// Set the transport so that all requests from httpClient will contain
 	// the collector credentials.
-	se.httpClient.Transport, err = se.RoundTripper(se.httpClient.Transport)
+	httpClient.Transport, err = se.RoundTripper(httpClient.Transport)
 	if err != nil {
-		return credentials.CollectorCredentials{},
-			registrationDone,
-			fmt.Errorf("couldn't create HTTP client transport: %w", err)
+		return nil, fmt.Errorf("couldn't create HTTP client transport: %w", err)
 	}
 
-	return colCreds, registrationDone, nil
+	return httpClient, nil
 }
 
-// Shutdown is invoked during service shutdown.
-func (se *SumologicExtension) Shutdown(ctx context.Context) error {
-	se.closeOnce.Do(func() { close(se.closeChan) })
-	select {
-	case <-ctx.Done():
-		return ctx.Err()
-	default:
-		return nil
-	}
-}
-
-// getCredentials returns the credentials either retrieved from local credentials
-// storage in cases they were available there or it registers the collector and
-// returns the credentials obtained from the API.
-// It also returns a boolean flag indicating if the successful registration
-// with the API took place and an error.
-func (se *SumologicExtension) getCredentials(ctx context.Context) (credentials.CollectorCredentials, bool, error) {
+// getCredentials retrieves the credentials for the collector.
+// It does so by checking the local credentials store and by validating those credentials.
+// In case they are invalid or are not available through local credentials store
+// then it tries to register the collector using the provided access keys.
+func (se *SumologicExtension) getCredentials(ctx context.Context, host component.Host) (credentials.CollectorCredentials, error) {
 	var (
-		colCreds         credentials.CollectorCredentials
-		registrationDone bool
-		err              error
+		colCreds credentials.CollectorCredentials
+		err      error
 	)
 
-	if se.credentialsStore.Check(se.hashKey) {
-		colCreds, err = se.credentialsStore.Get(se.hashKey)
-		if err != nil {
-			return credentials.CollectorCredentials{}, false, err
-		}
-		se.collectorName = colCreds.CollectorName
-		if !se.conf.Clobber {
-			if colCreds.ApiBaseUrl != "" {
-				se.baseUrl = colCreds.ApiBaseUrl
+	if !se.conf.Clobber {
+		colCreds, err = se.getLocalCredentials(ctx)
+		if err == nil {
+			if errI := se.injectCredentials(colCreds, host); errI != nil {
+				return credentials.CollectorCredentials{}, errI
+			}
```

Comment:
With the changes to `.validateCredentials`, can we now move this call to after the credentials were validated?

Reply:
What we could do is we could remove `injectCredentials` call and move it to `validateCredentials` as we'll need to set the `.registrationInfo` field anyway.

Reply:
I've moved stuff around PTAL @swiatekm-sumo